plugin/DirDiff.vim

```diff
@@ -616,12 +621,22 @@ function! <SID>CloseDiffWindows()
         wincmd k
         " Ask the user to save if buffer is modified
         call <SID>AskIfModified()
-        bd!
+        if g:DirDiffBufferDelete
+            bd!
+        else
+            diffoff
+            q
```

Comment:
I think this isn't a map, but an ex command (as `diffoff` and `bd`). To trigger a map a `normal!` command could be used. Talking about the maps, those could be change to the `noremap` version to ensure that no recursion could ever occur.